src/gpg/contact/delete.py

```diff
@@ -0,0 +1,30 @@
+import gpg
+import sys
+import os
+from fn.print_key import print_key
+
+
+def delete(contacts, force):
+    try:
+        c = gpg.Context()
+        for contact in contacts:
+            keys = list(c.keylist(contact))
+            for key in keys:
+                if(not force):
+                    print_key(key.fpr)
+                    ans = input("Delete this key from the keyring? (y/N)")
```

Comment:
I am currently somewhat imitating what egpg did earlier. This is one of the ways it could be done. I pushed it here to discuss more about it. 

So here is what gpg commands (egpg) originally did:
- If I pass a `egpg delete contact1` without --force
  - For each key of the contact gpg asked if you want to delete the key (with the exact same message)
 - it exited as soon as one key was deleted
- If I pass `egpg delete contact1 --force`
  - it deletes all keys matching contact1

What i am doing right now is:
- delete all keys matching the contact in case of `--force` without any messages
- for all matching keys with the contact ask one by one and let the user choose to delete or not `(y/N)` (I feel it is helpful if he wants to delete specific keys corresponding to a contact name). 

Would like your suggestions.

Reply:
Another way to approach this could be:

- by default doing a `cmd_contact_list contact1` first and asking `delete all these keys matching contact1 (y/N)`. If these are more keys let the user him/herself delete these by using keyids.

- we might have a option --interactive which has the current behavior as mentioned in the above comment.

Reply:
When we talk about `contact1`, as an argument to the delete command, it can be an email address, or a full name, or a first name, etc. So, it is actually a search pattern which may match more than one contact (called `key` in the gpg language, which is an overloaded term according to me). It makes sense to ask for each of these matching contacts whether they should be deleted or not. But when we delete a contact, we delete it completely, not only one subkey of it. I believe this is also the current behavior of `gpg`.

Reply:
Yup I think I got it : a contact in egpg is identified as a key (with subkeys) in gpg / gpgme.

I am deleting all subkeys with the key at once only. 
I was actually talking about the case of multiple contact matches occur corresponding to the string passed.

Reply:
I had borrowed the message `Delete this key from the keyring? (y/N)` from gpg only. If it seems confusing I can change `key` to `contact`. 
It does delete a key with all subkeys as mentioned above.

Reply:
Yes, it should be confusing for the user, because as far as the user knows there is only one key, his own private key. All the rest are contacts.